tests/test_processor.py

```diff
@@ -328,3 +400,120 @@ def test_mixed_deps_two_apps(mock_repo_file, processor, optional_deps_method, ex
     processor.requested_app_names = ["app1", "app3"]
     processed = processor.process()
     assert_clowdapps(processed["items"], expected)
+
+
+# Testing --no-remove-resources/dependency "app:" syntax
+def test_should_remove_remove_for_none_no_exceptions():
+    remove_resources = AppOrComponentSelector(select_all=False, components=[], apps=[])
```

Comment:
Can we add a comment showing the CLI options "as they would be seen on the CLI" for each test

```suggestion
# --no-remove-resources all --no-remove-resources component1 --no-remove-resources app1
remove_resources = AppOrComponentSelector(select_all=False, components=[], apps=[])
```